backend/Recodo.BLL/Services/AccessToVideoService.cs

```diff
@@ -0,0 +1,87 @@
+using Recodo.BLL.Services.Abstract;
+using Recodo.DAL.Context;
+using Recodo.DAL.Entities;
+using AutoMapper;
+using System.Threading.Tasks;
+using Recodo.Common.Dtos.Access;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace Recodo.BLL.Services
+{
+    public class AccessToVideoService : BaseService
+    {
+        private UserService _userService;
+        public AccessToVideoService(RecodoDbContext context, IMapper mapper, UserService userService)
+            : base(context, mapper)
+        {
+            _userService = userService;
+        }
+
+        public async Task AddUserAccess(string email, int videoId)
+        {
+            var unRegisteredUser = new AccessForUnregisteredUsersDTO
+            {
+                Email = email,
+                VideoId = videoId
+            };            
+            var user = await _userService.GetUserByEmail(email);
+            var accessedUser = await FindUnregisteredUser(unRegisteredUser);
```

Comment:
I'd change the flow a bit. No sense to create DTO, look in the context if there's a registered user. Do this only in case you didn't find a registered user